nullaway/src/main/java/com/uber/nullaway/handlers/InferredJARModelsHandler.java

```diff
@@ -109,21 +101,24 @@ public InferredJARModelsHandler(Config config) {
     }
   }
 
-  /*
-   * Scan Java Classpath for JarInfer model jars and map their names to locations
+  /**
+   * Loads all stubx files discovered in the classpath. Stubx files are discovered via
+   * implementations of {@link JarInferStubxProvider} loaded using a {@link ServiceLoader}
    */
-  private void processClassPath() {
-    URL[] classLoaderUrls =
-        ((URLClassLoader) Thread.currentThread().getContextClassLoader()).getURLs();
-    for (URL url : classLoaderUrls) {
-      String path = url.getFile();
-      if (path.matches(config.getJarInferRegexStripModelJarName())) {
-        String name = path.replaceAll(config.getJarInferRegexStripModelJarName(), "$1");
-        LOG(DEBUG, "DEBUG", "model jar name: " + name + "\tjar path: " + path);
-        if (!mapModelJarLocations.containsKey(name)) {
-          mapModelJarLocations.put(name, new LinkedHashSet<>());
+  private void loadStubxFiles() {
+    Iterable<JarInferStubxProvider> astubxProviders =
+        ServiceLoader.load(
+            JarInferStubxProvider.class, InferredJARModelsHandler.class.getClassLoader());
+    for (JarInferStubxProvider provider : astubxProviders) {
+      for (String astubxPath : provider.pathsToStubxFiles()) {
+        Class<? extends JarInferStubxProvider> providerClass = provider.getClass();
+        InputStream stubxInputStream = providerClass.getResourceAsStream(astubxPath);
```

Comment:
Now loading using the provider's class rather than `this.getClass()`.  This has the added benefit of allowing the provider's returned paths to be relative to its location